src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -329,29 +335,61 @@ private boolean intersectionsAreExplicit(final GeometricSurface waterbody,
                 .haveExplicitLocationsForIntersections(polygon, lineItem));
     }
 
+    /**
+     * If the waterbody is an area, the intersection is valid. If it's not an area, it's a
+     * multipolygon, and we make sure that either 1. The intersection is at an outer member of the
+     * multipolygon and the outer member was not synthetically created by the atlas slicing process.
+     * It's estimated that if the outer member doesn't have the {@link LastEditTimeTag}, it likely
+     * did not come from an OSM feature, which suggests it was created due to atlas slicing and
+     * would therefore not be a geometrically accurate member of the relation. 2. The intersecting
+     * feature is entirely floating in an outer member of the relation, and not
+     * intersecting/floating in any of the inner members. The inner members are typically islands so
+     * we avoid flagging in the latter scenario. The outer member must still have the LastEditTime
+     * tag
+     *
+     * @param waterbody
+     *            the GeometricSurface representation of the waterbody
+     * @param object
+     *            the AtlasEntity representation of the waterbody
+     * @param intersectingFeature
+     *            the intersecting building/road/line
+     * @return true if either of the 2 above conditions are satisfied, false otherwise
+     */
+    private boolean intersectsValidPartOfWaterbody(final GeometricSurface waterbody,
+            final AtlasObject object, final PolyLine intersectingFeature)
+    {
+        return !(waterbody instanceof MultiPolygon) || ((Relation) object).members().stream()
+                .filter(member -> member.getRole().equals("outer"))
+                .map(member -> new Tuple<>(member.getEntity(),
+                        new Polygon(member.getEntity() instanceof Area ? (Area) member.getEntity()
+                                : (LineItem) member.getEntity())))
+                .anyMatch(member -> (intersectingFeature.intersects(member.getSecond())
+                        || (intersectingFeature.within(member.getSecond())
+                                && ((MultiPolygon) waterbody).inners().stream().noneMatch(
+                                        innerMember -> intersectingFeature.intersects(innerMember)
+                                                || intersectingFeature.within(innerMember))))
+                        && member.getFirst().lastEdit().isPresent());
```

Comment:
While this seems like it works -- this solution feels a bit hacky. We've ran into this issue before and proposed parsing the identifier to verify whether or not the feature has been sliced at the border -- this too felt like a hack. 
I would defer to atlas to provide a better (reusable) solution to verify if a feature has been country sliced -- as its shown to be useful in other checks.

Reply:
I agree that this should definitely be in Atlas. However, just to get a short term solution/discussion going, I did a minor change on this logic to be more in line with how country slicing works- we're now making sure the relation member has any tag other than the IsoCountryTag and SyntheticNearestNeighborCountryCodeTag. These two tags are synthetically added to a relation member that's been sliced at the border. The new logic should validate the same features as the previous method since all validated members should still have the LastEditTimeTag/other OSM specific tags.

This change is in 28a3ce6